File: calico-us-berkeley/2025-fall/final/chess/templates/chess_template.cpp
#include <iostream>
#include <tuple>
#include <vector>

using namespace std;

int solve(int N, int M, int K, vector<tuple<char, int, int>> P) {
    // N: Number of rows of the board
    // M: Number of columns of the board
    // K: Number of pieces
    // P: list of tuples. Each tuple contains (piece type, x position, y position)

    // YOUR CODE HERE
    return 0;
}

int main() {
	int t; cin >> t;
	for (int i = 0; i < t; i++) {
		int n, m; cin >> n >> m;
		int k; cin >> k;
		vector<tuple<char, int, int>> pieces;
		while (k--) {
			char piece; int x, y; cin >> piece >> x >> y;
            pieces.emplace_back(piece, x, y);
		}
	    cout << solve(n, m, k, pieces) << "\n";
	}
}

File: calico-us-berkeley/2025-fall/final/chess/templates/chess_template.py
def solve(N: int, M: int, K: int, P: list[tuple[str, int, int]]) -> int:
    """
    
    N: Number of rows of the board
    M: Number of columns of the board
    K: Number of pieces
    P: list of tuples. Each tuple contains (piece type, x position, y position)
    """
    # YOUR CODE HERE
    return 0


def main():
    T = int(input())
    for _ in range(T):
        temp = input().split()
        N, M = int(temp[0]), int(temp[1])
        K = int(input())
        P = []
        for _ in range(K):
            temp = input().split()
            temp_tuple = (str(temp[0]), int(temp[1]), int(temp[2]))
            P.append(temp_tuple)
        print(solve(N, M, K, P))

if __name__ == '__main__':
    main()
